src/Toolkit/Toolkit/UI/Controls/BasemapGallery/BasemapGalleryItem.cs

```diff
@@ -246,7 +246,8 @@ internal bool EqualsBasemap(Basemap? other)
                 return false;
             }
 
-            return other == Basemap || other.Name == Basemap?.Name
+            return other == Basemap || (Basemap.BaseLayers.Count == other.BaseLayers.Count
+                                        && Basemap.BaseLayers.All(layer => other?.BaseLayers.FirstOrDefault(l => l.Name == layer.Name) is not null))
                                     || (other.Item?.ItemId != null && other.Item?.ItemId == Basemap?.Item?.ItemId)
                                     || (other.Uri != null && other.Uri == Basemap?.Uri);
```

Comment:
We should do early exit on these simple more rigid checks first. If if itemid is the same, just return right away, rather than doing all the layer comparison up-front.
Don't check Layer.Name but instead use the Source URI to compare as well as type comparisons. This entire expression is basically gonna be a lot more complicated than before.
You also forgot to check the ReferenceLayers collection on the basemap.

Reply:
I thought we are already doing URI comparisons with `(other.Uri != null && other.Uri == Basemap?.Uri)`. Would it be helpful to check `Layer.Id`?

Reply:
I mean the URL of the layers. A basemap doesn't have a URI when it's constructed manually like you were doing in your test

Reply:
[Layer](https://developers.arcgis.com/net/api-reference/api/netwin/Esri.ArcGISRuntime/Esri.ArcGISRuntime.Mapping.Layer.html) type doesn't seems to have URL property until it is casted to specific Layer Type.